demo-app/src/main/kotlin/io/getstream/video/android/DeeplinkingActivity.kt

```diff
@@ -72,8 +72,9 @@ class DeeplinkingActivity : ComponentActivity() {
             }
         }
 
+        val callIdFromExtra = intent?.getStringExtra(CALL_ID)
         val data: Uri = intent?.data ?: return
-        val callId = data.getQueryParameter("id") ?: return
+        val callId = callIdFromExtra ?: extractCallId(data) ?: return
```

Comment:
We need to remove these two `return` here - they cause the activity to just show an endless loading if this is actually null and it's not obvious what is wrong.

I suggest something like this:

```
        if (data == null) {
            logger.e { "Can't open the call from deeplink because intent data is null" }
            finish()
            return
        }


        if (callId == null) {
            logger.e { "Can't open the call from deeplink because call ID is null" }
            finish()
            return
        }
```